### ISSUE TITLE
Staking Partial-Reward Duration (aka Warmup Period)

### ISSUE THREAD

**Issue by slavakurilyak** [net2devcrypto/nftstaking#3]
As a contract owner, I want to implement a warmup period for NFT staking (aka stake NFTs at 50% of reward value for 7 days, 10 days, or 30 days) so that I can prevent bad actors from quickly acquiring enough stakes in the project and incentivize long-term staking

See @solana-labs's [staking lifecycle](https://docs.solana.com/cluster/stake-delegation-and-rewards#stake-warmup-cooldown-withdrawal) for inspiration

**Comment by net2devcrypto** [comment]
> As a contract owner, I want to implement a warmup period for NFT staking (aka stake NFTs at 50% of reward value for 7 days, 10 days, or 30 days) so that I can prevent bad actors from quickly acquiring enough stakes in the project and incentivize long-term staking
> 
> 
> 
> See @solana-labs's [staking lifecycle](https://docs.solana.com/cluster/stake-delegation-and-rewards#stake-warmup-cooldown-withdrawal) for inspiration

Yep this is very common, let me review the code and come up with timelocking on the contract.
